### ISSUE TITLE
Upgrade to Async API 3.0?

### ISSUE THREAD

**Issue by robert-sykes** [Ferror/asyncapi-doc-bundle#7]
Just looking through the code, I see we are aiming at Async API 2.6.0. I don't think 3.0 adds a huge amount to the schema, I'd have to have a look over the changes, but thought I'd check in to see if this is something we want to do?

We could add an addition that allows for selecting a version and generate the schema on that, i.e. Version 2, Version 3 on generate to output the appropriate schema. 

Just thought dumping.

**Comment by Ferror** [comment]
Yeah, Async API 3.0 brings some major changes. I think separating concerns we should abstract the Schema class: https://github.com/Ferror/asyncapi-doc-bundle/blob/master/src/Schema.php

**Comment by robert-sykes** [comment]
I agree, we can have an interface an implement version specific schemas. I imagine the Schema Generator may need abstracting too?

**Comment by Ferror** [comment]
@robert-sykes I'm starting to build the Schema V3 generator. I need kinda to figure out the Attributes to it does not depend on Channel Type

**Comment by robert-sykes** [comment]
@Ferror Sounds good - i've had a quick look over the related PR, it's looking good. I'll have a think about the Attributes when I get a chance to give this a proper look.

**Comment by Ferror** [comment]
@robert-sykes, I made the current setup work with Async API v3. It's not ideal, as it does not cover Operations yet, but it enables us to define attributes without thinking about the Schema Version. Feel free to code review PR's post-factum. I made sure created by application schemas are valid for both versions - GitHub action matrix 😄

**Comment by robert-sykes** [comment]
@Ferror Hi fella, sorry, I am just coming back all this after working on a few unrelated projects. Whats the current status on the V3 Update? I can review the PR if it's ready.

**Comment by Ferror** [comment]
@robert-sykes I remember something is still missing. I will sit down and finish it during the May Holidays 😄
